src/coral_models/compute_metrics.py

```diff
@@ -22,34 +22,46 @@ def compute_wer_metrics(pred: EvalPrediction, processor: Processor) -> dict[str,
             dictionary with 'wer' as the key and the word error rate as the value.
     """
     wer_metric = load_metric("wer")
-    pad_token = processor.tokenizer.pad_token_id
+    tokenizer: PreTrainedTokenizerBase = getattr(processor, "tokenizer")
+    pad_token = tokenizer.pad_token_id
 
-    # Shape: [batch_size, seq_len, vocab_size]
+    # Shape: [batch_size, seq_len, vocab_size] or [batch_size, seq_len]
     predictions: NDArray[np.int_] | NDArray[np.float_] = pred.predictions
 
-    # In cases where the prediction logits are -100 for all characters, convert the
-    # logits for the padding character to be 100, to make sure that padding is chosen
-    # TODO: Does this actually do what we want it to do?
-    pad_arr = predictions[:, :, pad_token]
-    predictions[:, :, pad_token] = np.where(pad_arr == -100, 100, pad_arr)
-
-    # Decode the predictions to get the transcriptions. When a language model is
-    # attached to the processor then we get the predicted string directly from the
-    # logits
-    if predictions.dtype == np.int_:
-        # If the vocabulary dimension of the predictions is too small then we pad with
-        # zeros to match the size of the vocabulary
-        vocab_size = processor.tokenizer.get_vocab()
-        mismatch_dim = len(vocab_size) - predictions.shape[-1]
-        predictions = np.pad(predictions, ((0, 0), (0, 0), (0, mismatch_dim)))
-
-        predictions_str = [processor.decode(token_ids) for token_ids in predictions]
-
-    # Otherwise, if we are not using a language model, we need to convert the logits to
-    # token IDs and then decode the token IDs to get the predicted string
+    if len(predictions.shape) == 3:
+        # In cases where the prediction logits are -100 for all characters, convert the
+        # logits for the padding character to be 100, to make sure that padding is
+        # chosen
+        # TODO: Does this actually do what we want it to do?
```

Comment:
These particular lines are still quite mysterious to me. As far as i can tell (caveat: only check simple examples) it does what your comment conveys, BUT i am super unsure about what the is so, its hard to verify. Quick talk when you get back?

Reply:
Removed it now. I remember I included it to deal with a rare error, but I'll just include it back in if it occurs again